components/advanced_create_post/advanced_create_post.tsx

```diff
@@ -1519,7 +1525,38 @@ class AdvancedCreatePost extends React.PureComponent<Props, State> {
         );
     }
 
+    isValidPersistentNotifications = (): boolean => {
+        if (!this.hasPrioritySet()) {
+            return true;
+        }
+
+        const {currentChannel} = this.props;
+        const {priority, persistent_notifications: persistentNotifications} = this.props.draft.metadata!.priority!;
+        if (priority !== PostPriority.URGENT || !persistentNotifications) {
+            return true;
+        }
+
+        if (currentChannel.type === Constants.DM_CHANNEL) {
+            return true;
+        }
+
+        if (this.hasSpecialMentions()) {
+            return false;
+        }
+
+        const mentions = mentionsMinusSpecialMentionsInText(this.state.message);
```

Comment:
Note that this function just counts things that look like mentions, so this will allow someone to enable persistent notifications for a message that just contains a fake mention for a user that doesn't exist. It also seems to capture mentions in things like code blocks that otherwise wouldn't trigger mentions

Reply:
This is only for a quick validation I didn't want to burden the `onChange` much.
We are validating usernames in the confirmation modal.

I didn't think of code blocks at all though, let me check on that.